pylint/checkers/misc.py

```diff
@@ -137,24 +97,47 @@ def _check_encoding(self, lineno, line, file_encoding):
                                       ' bad encoding'.format(file_encoding))
 
     def process_module(self, module):
-        """inspect the source file to find encoding problem or fixmes like
-        notes
-        """
-        if self.config.notes:
-            notes = re.compile(
-                r'#\s*(%s)\b' % "|".join(map(re.escape, self.config.notes)), re.I)
-        else:
-            notes = None
+        """inspect the source file to find encoding problem"""
         if module.file_encoding:
             encoding = module.file_encoding
         else:
             encoding = 'ascii'
 
         with module.stream() as stream:
             for lineno, line in enumerate(stream):
-                line = self._check_encoding(lineno + 1, line, encoding)
-                if line is not None and notes:
-                    self._check_note(notes, lineno + 1, line, module.tolineno)
+                self._check_encoding(lineno + 1, line, encoding)
+
+    def process_tokens(self, tokens):
+        """inspect the source to find fixme problems"""
+        if not self.config.notes:
+            return
+        comments = [token_info for token_info in tokens
+                    if token_info.type == tokenize.COMMENT]
+        for comment in comments:
+            comment_text = comment.string[1:].lstrip()  # trim '#' and whitespaces
+
+            # handle pylint disable clauses
+            disable_option_match = OPTION_RGX.search(comment_text)
+            if disable_option_match:
+                try:
+                    _, value = disable_option_match.group(1).split('=', 1)
+                    values = [_val.strip().upper() for _val in value.split(',')]
+                    if set(values) & set(self.config.notes):
+                        return
+                except ValueError:
+                    self.add_message('bad-inline-option',
+                                     args=disable_option_match.group(1).strip(),
+                                     line=comment.string)
+                    return
+
+            # emit warnings if necessary
+            for note in self.config.notes:
+                if comment_text.lower().startswith(note.lower()):
+                    self.add_message('fixme',
```

Comment:
We can do a break after this `add_message` as it's not possible for the comment to start with other notes after this step.